src/test/java/gov/healthit/chpl/aqa/stepDefinitions/Hooks.java

```diff
@@ -1,147 +1,166 @@
-     */
-    public static void takeScreenshot(final String hash) throws Exception {
-        File scrFile = ((TakesScreenshot) getDriver()).getScreenshotAs(OutputType.FILE);
-        File outFile = new File(screenshotPath + File.separator + "failed-test-" + hash.replaceAll("\\W+", "_") + ".png");
-        FileUtils.copyFile(scrFile, outFile);
-    }
-
-    /**
-     * Take a screenshot. Generates hash based on current time.
-     * @throws Exception if screenshot cannot be created
-     */
-    public static void takeScreenshot() throws Exception {
-        String hash = String.valueOf(new Date().getTime());
-        takeScreenshot(hash);
-    }
-    
-
-}
+		 */
+
+		if(System.getenv("browser")!=null && !System.getenv("browser").isEmpty()){
+			browser = System.getenv("browser");
+			System.setProperty("webdriver.gecko.driver", System.getProperty("PathtoGeckodriver") + File.separator + "geckodriver");
+
+		}else{
+			fis = new FileInputStream(
+					System.getProperty("user.dir") + File.separator+"src"
+							+File.separator+"test"+File.separator+"resources"
+							+File.separator+"Config.properties");
+			config.load(fis);
+			browser = config.getProperty("browser");
+
+			System.setProperty("webdriver.gecko.driver", System.getProperty("PathtoGeckodriver") + File.separator + "geckodriver.exe");
+
+		}
+		config.setProperty("browser", browser);
+		if (StringUtils.isEmpty(downloadPath)) {
+			String tempDirectory;
+			try {
+				tempDirectory = Files.createTempDirectory("download-files").toString();
+				// Print the path to the newly created directory
+			} catch (final IOException e) {
+				// If temp directory creation failed, create new directory in target folder
+				// user.dir - User working directory, make new directories in user's working directory
+				File file = new File("target", "download-files");
+				file.mkdirs();
+				tempDirectory = System.getProperty("user.dir") + File.separator + "target" + File.separator + "download-files";
+			}
+			downloadPath = tempDirectory;
+		}
+		dir = new File(downloadPath);
+		if (!dir.exists()) {
+			dir.mkdirs();
+		}
+
+		if (config.getProperty("browser").equals("chrome")) {
+			HashMap<String, Object> chromePrefs = new HashMap<String, Object>();
+			chromePrefs.put("profile.default_content_settings.popups", 0);
+			chromePrefs.put("download.default_directory", downloadPath);
+			/**
+			 * Save Chrome Options
+			 */
+			ChromeOptions options = new ChromeOptions();
+			options.setExperimentalOption("prefs", chromePrefs);
+			chromePrefs.put("safebrowsing.enabled", "true");
+			options.addArguments("--safebrowsing-disable-download-protection");
+			options.addArguments("disable-popup-blocking");
+			options.setCapability(CapabilityType.ACCEPT_SSL_CERTS, true);
+			screenshotPath = System.getProperty("user.dir") + File.separator + "test-output";
+			driver = new EventFiringWebDriver(new ChromeDriver(options));
+		}
+
+		else if(config.getProperty("browser").equals("firefox")) {
```

Comment:
This line should be on line 104, as `} else if (config...`, and checkstyle should have warned about this